src/util/animate.js

```diff
@@ -171,29 +171,29 @@
             valuePerc = isMany ? Math.abs((current[0] - startValue[0]) / byValue[0])
               : Math.abs((current - startValue) / byValue);
         //  update context
-        context.currentValue = current;
+        context.currentValue = isMany ? current.slice() : current;
         context.completionRate = valuePerc;
         context.durationRate = timePerc;
         if (cancel) {
           return;
         }
-        if (abort(current, valuePerc, timePerc)) {
+        if (abort(isMany ? current.slice() : current, valuePerc, timePerc)) {
           removeFromRegistry();
           return;
         }
         if (time > finish) {
           //  update context
-          context.currentValue = endValue;
+          context.currentValue = isMany ? endValue.slice() : endValue;
           context.completionRate = 1;
           context.durationRate = 1;
           //  execute callbacks
-          onChange(endValue, 1, 1);
-          onComplete(endValue, 1, 1);
+          onChange(isMany ? endValue.slice() : endValue, 1, 1);
+          onComplete(isMany ? endValue.slice() : endValue, 1, 1);
```

Comment:
here we need to slice only once
and the second argument needs to be any array as well in case of `isMany` I think.

Reply:
In this case we need separate because mutating the value (if shared) will mutate for the other (`onComplete`)

Reply:
```suggestion
          onChange(isMany ? endValue.slice() : endValue, 1, 1);
          onComplete(endValue, 1, 1);
```

Isn't this enough?

Reply:
?